code/modules/research/xenobiology/crossbreeding/warping.dm

```diff
@@ -0,0 +1,1135 @@
+
+
+///adds the jolly mood effect along with hug sound effect.
+/obj/effect/warped_rune/pinkspace/Crossed(atom/movable/crossing)
+	. = ..()
+	if(istype(crossing,/mob/living/carbon/human))
+		playsound(rune_turf, "sound/weapons/thudswoosh.ogg", 50, TRUE)
+		SEND_SIGNAL(crossing, COMSIG_ADD_MOOD_EVENT,"jolly", /datum/mood_event/jolly)
+		crossing.visible_message(crossing, "<span class='notice'>You feel happier.</span>")
+
+
+/*Gold rune : Turn things over it to gold, completely fucks over economy */
+
+/obj/item/slimecross/warping/gold
+	colour = "gold"
+	runepath = /obj/effect/warped_rune/goldspace
+	effect_desc = "Draws a rune capable of turning nearly anything put on it to gold every minute"
+
+
+/obj/effect/warped_rune/goldspace
+	icon_state = "midas_rune"
+	desc = "When everything is abundant, it becomes easy to forget when one had to work to obtain anything."
+	max_cooldown = 600
+
+
+/obj/effect/warped_rune/goldspace/Initialize()
+	. = ..()
+	START_PROCESSING(SSprocessing, src)
+
+
+///turn anything on the tile to a golden version of themselves. Sheets of any kind will be directly turned into gold bars.
+/obj/effect/warped_rune/goldspace/process()
+	if(cooldown > world.time)
+		return
+
+	for(var/obj/item/goldened in rune_turf)
+		if(istype(goldened,/obj/item/stack/sheet/mineral/gold))//can't turn gold into gold
+			return
+
+		var/gold_amount = 0
+		for(var/mats in goldened.custom_materials)
+			gold_amount += goldened.custom_materials[mats]
+		if(!gold_amount)
+			gold_amount = 50 //if the item doesn't have any material it makes the item worth around 0.005 gold bars.
+		goldened.material_flags = MATERIAL_COLOR | MATERIAL_ADD_PREFIX | MATERIAL_AFFECT_STATISTICS
+		goldened.set_custom_materials(list(/datum/material/gold = gold_amount))
+		cooldown = world.time + max_cooldown //only set the cooldown once something's been turned to gold
+		return
+
+
+/*Adamantine rune, will spawn ores depending on the mineral rocks surrounding it. Here to make miners do their job even less.  */
+
+
+/obj/item/slimecross/warping/adamantine
+	colour = "adamantine"
+	runepath = /obj/effect/warped_rune/adamantinespace
+	effect_desc = "draws a rune capable of copying the ores of nearby mineral rocks."
+
+
+/obj/effect/warped_rune/adamantinespace  //doesn't have a rune icon yet please spriters help me I can't sprite for shit I beg you
+	desc = "The universe's ressource are nothing but tools for us to use and abuse."
+	max_cooldown = 300  //"mines" things every 30 seconds.
+	icon_state = "mining_rune"
+
+
+/obj/effect/warped_rune/adamantinespace/Initialize()
+	. = ..()
+	START_PROCESSING(SSprocessing, src)
+
+
+/obj/effect/warped_rune/adamantinespace/process()
+	if(cooldown < world.time)
+		cooldown = world.time + max_cooldown
+		auto_mining()
+
+
+/obj/effect/warped_rune/adamantinespace/proc/auto_mining()
+	for(var/turf/closed/mineral/ores in range(7,rune_turf)) //the range is pretty big to at least try to rival miners and their plasma cutters.
```

Comment:
the large range is actually a detriment, as in order to cherry pick the ores this thing will harvest, you're gonna have to remove all other kinds of ore from this range

Reply:
Because it will do that for each mineral in range, it will still copy ALL ores in range, not just the first one it comes across, so if you have 10 diamond vein in range it will produce 10 diamond ores for exemple.

Reply:
ohhhhhhhhhh